src/lxc/pam/pam_cgfs.c

```diff
@@ -223,6 +223,17 @@ static bool cgv2_prune_empty_cgroups(const char *user);
 static bool cgv2_remove(const char *cgroup);
 static bool is_cgv2(char *line);
 
+static int do_mkdir(const char *path, mode_t mode)
+{
+        mode_t mask;
+        int r;
+
+        mask = umask(0);
+        r = mkdir(path, mode);
```

Comment:
Looks fine - though how about keeping and resetting saved_errno?

Reply:
Yeah might as well, although `umask` doesn't touch it, does it?

Reply:
Well, POSIX doesn't give a guarantee ;)

Reply:
Fair enough, I pushed the change + rebased on master